mitiq/rem/rem.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (C) 2021 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Readout Confusion Inversion."""
+
+from typing import Callable, Optional, Union
+
+from functools import wraps
+import numpy as np
+import numpy.typing as npt
+
+from mitiq._typing import QPROGRAM, MeasurementResult
+from mitiq.executor.executor import Executor
+from mitiq.observable.observable import Observable
+from mitiq.rem.inverse_confusion_matrix import mitigate_measurements
+
+
+def execute_with_rem(
+    circuit: QPROGRAM,
+    executor: Union[Executor, Callable[[QPROGRAM], MeasurementResult]],
+    observable: Optional[Observable] = None,
+    *,
+    inverse_confusion_matrix: npt.NDArray[np.float64],
+) -> float:
```

Comment:
I only recently learning about this `*` syntax to make arguments keyword only! Very cool. Just out of curiosity, is there a reason you want `inverse_confusion_matrix` to be keyword only?

Reply:
@natestemen, sorry to step in, but I asked @amirebrahimi to make `inverse_confusion_matrix`  a keyword-only argument.
The reason is for consistency with all other `execute_with_xxx` functions, e.g. PEC and DDD. The motivation is that we want `observable` to be the third positional argument but with a default value of `None`. 
This fact forces all the arguments on the right side to be keyword-only.

Another reason is that keyword-only arguments are great to avoid position mistakes, especially when there are many options. But this is just my opinion and I am not sure if it is a best practice.

Reply:
I'm in favor of it! I was just curious if there was a reason we used it here. Thanks for the info.